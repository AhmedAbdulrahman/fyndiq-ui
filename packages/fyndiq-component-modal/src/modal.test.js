import React from 'react'
import { shallow } from 'enzyme'
import Modal from './modal'

describe('Modal Component', () => {
  it('should return null if not open', () => {
    expect(shallow(<Modal />).isEmptyRender()).toBe(true)
  })

  it('should be displayed correctly if opened', () => {
    expect(shallow(<Modal open>Content</Modal>)).toMatchSnapshot()
  })

  it('should have custom classes for different elements', () => {
    expect(
      shallow(
        <Modal
          overlayClassName="overlayClassName"
          wrapperClassName="wrapperClassName"
          closeClassName="closeClassName"
          open
        >
          Content
        </Modal>,
      ),
    ).toMatchSnapshot()
  })

  it('should call onClose when clicking on the overlay', () => {
    const spy = jest.fn()
    const Component = shallow(<Modal onClose={spy} open />)
    Component.find('.overlay').simulate('click')
    expect(spy).toHaveBeenCalled()
  })

  it('should call onClose when clicking on the x', () => {
    const spy = jest.fn()
    const Component = shallow(<Modal onClose={spy} open />)
    Component.find('.close').simulate('click')
    expect(spy).toHaveBeenCalled()
  })

  it('should stop the propagation when clicking on the wrapper', () => {
    const stopSpy = jest.fn()
    const Component = shallow(<Modal open />)
    Component.find('.wrapper').simulate('click', { stopPropagation: stopSpy })
    expect(stopSpy).toHaveBeenCalled()
  })

  it('should call onClose when pressing ESC', () => {
    const spy = jest.fn()
    shallow(<Modal onClose={spy} open />)
    global.simulate.keypress({ keyCode: 27 })
    expect(spy).toHaveBeenCalled()
  })

  it('should unregister keypress event when unmounting', () => {
    const Component = shallow(<Modal open />)
    Component.unmount()
    expect(global.simulate.keypress).toBe(undefined)
  })

  it('should handle function children', () => {
    const spy = jest.fn()
    const Component = shallow(
      <Modal open onClose={spy}>
        {({ onClose }) => (
          <button className="custom-close" onClick={onClose}>
            Button
          </button>
        )}
      </Modal>,
    )

    Component.find('.custom-close').simulate('click')
    expect(spy).toHaveBeenCalled()
  })
})
