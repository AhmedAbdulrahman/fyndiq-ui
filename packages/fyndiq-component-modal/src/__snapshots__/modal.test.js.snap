// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should be displayed correctly if opened 1`] = `
<ModalPortal
  bodyLock={true}
  portalId="fyndiq-modal"
>
  <div
    className="overlay "
    onClick={[Function]}
  >
    <div
      className="wrapper "
      onClick={[Function]}
    >
      <button
        className="close "
        onClick={[Function]}
      >
        ×
      </button>
      Content
    </div>
  </div>
</ModalPortal>
`;

exports[`Modal Component should have custom classes for different elements 1`] = `
<ModalPortal
  bodyLock={true}
  portalId="fyndiq-modal"
>
  <div
    className="overlay overlayClassName"
    onClick={[Function]}
  >
    <div
      className="wrapper wrapperClassName"
      onClick={[Function]}
    >
      <button
        className="close closeClassName"
        onClick={[Function]}
      >
        ×
      </button>
      Content
    </div>
  </div>
</ModalPortal>
`;
